usecases/webhook_events_usecase.go

```diff
@@ -77,28 +90,48 @@ func (usecase WebhookEventsUsecase) CreateWebhookEvent(
 		return err
 	}
 
-	webhookEventId := uuid.New().String()
-	err = usecase.webhookEventsRepository.CreateWebhookEvent(ctx,
-		tx,
-		webhookEventId,
-		input,
-	)
+	err = usecase.webhookEventsRepository.CreateWebhookEvent(ctx, tx, input)
 	if err != nil {
 		return errors.Wrap(err, "error creating webhook event")
 	}
-
 	return nil
 }
 
-func (usecase WebhookEventsUsecase) SendWebhookEvents(
+// SendWebhookEventAsync sends a webhook event asynchronously, with a new context and timeout and a child span.
+func (usecase WebhookEventsUsecase) SendWebhookEventAsync(ctx context.Context, webhookEventId string) {
+	logger := utils.LoggerFromContext(ctx).With("webhook_event_id", webhookEventId)
+	ctx = utils.StoreLoggerInContext(ctx, logger)
+
+	// go routine to send the webhook event asynchronously, with a new context and timeout and a child span
+	go func() {
+		ctx, cancel := context.WithTimeout(context.WithoutCancel(ctx), time.Second*3)
+		defer cancel()
+		tracer := utils.OpenTelemetryTracerFromContext(ctx)
+		ctx, span := tracer.Start(
+			ctx,
+			"CreateWebhookEvent.SendWebhookEventAsync",
+			trace.WithAttributes(attribute.String("webhook_event_id", webhookEventId)))
+		defer span.End()
+
+		status, err := usecase._sendWebhookEvent(ctx, webhookEventId)
+		if err != nil {
+			logger.ErrorContext(ctx, fmt.Sprintf("Error sending webhook event %s: %s", webhookEventId, err.Error()))
+		}
+		if status == models.Retry {
+			logger.WarnContext(ctx, fmt.Sprintf("Webhook event %s will be retried", webhookEventId))
+		}
```

Comment:
IMO, not really accurate: normally we do not use returned value when `err != nil`, here by reading the `_sendWebhookEvent` implementation, there is some branching where the returned status may not be the one currently present in the DB

Maybe use a `else` or add a `return` in previous `if`

Reply:
Ok, thanks. Took me a while to understand your point but i'm there now.
Yes, it is correct that I'm breaking a convention here - partly because I'm mixing up the error from sending the webhook and the error from udpating the webhook event status in db